tasks/image-segmentation/src/validation.py

```diff
@@ -0,0 +1,192 @@
+from pathlib import Path
+
+import logging
+import csv
+
+from coretex import TaskRun, ImageSample, ImageDatasetClasses, Model, folder_manager
+from coretex.utils import resizeWithPadding
+
+import numpy as np
+import tensorflow as tf
+import matplotlib.pyplot as plt
+
+from src.utils import hasDotAnnotation
+
+
+def meanIouScore(iouScores: list[float]) -> float:
+    return sum(iouScores) / len(iouScores)
+
+
+def plotImageSegmentation(realImage: np.ndarray, trueSegmentation: np.ndarray, predictedSeqmentation: np.ndarray, iouScore: float, name: str, taskRun: TaskRun) -> None:
+    fig, axes = plt.subplots(1, 3, figsize = (15, 5))
+
+    axes[0].imshow(realImage, cmap = "summer")
+    axes[0].set_title("Original image")
+    axes[0].axis("off")
+
+    axes[1].imshow(trueSegmentation, cmap = "summer")
+    axes[1].set_title("True segmentation")
+    axes[1].axis("off")
+
+    axes[2].imshow(predictedSeqmentation, cmap = "summer")
+    axes[2].set_title(f"Predicted segmentation\n IoU Score: {iouScore}")
+    axes[2].axis("off")
+
+    plt.savefig(f"{folder_manager.temp / name}.jpg")
+    taskRun.createArtifact(f"{folder_manager.temp / name}.jpg", f"images/{name}.jpg")
+    plt.close()
+
+
+def iouScoreClass(testMask: np.ndarray, predictedMask: np.ndarray) -> float:
+    intersecrion = np.logical_and(testMask, predictedMask)
+    union = np.logical_or(testMask, predictedMask)
+
+    return float(np.sum(intersecrion) / np.sum(union))
+
+
+def iouScoreImage(testMask: np.ndarray, predictedMask: np.ndarray) -> float:
+    iouScores: list[float] = []
+
+    classLabels = np.unique(testMask)
+    if np.any(classLabels == 0) and len(classLabels) == 1:
+        return 0.0
+
+    for label in classLabels:
+        if label > 0:
+            testMaskClass = testMask == label
+            predictedMaskClass = predictedMask == label
+            iouScores.append(iouScoreClass(testMaskClass, predictedMaskClass))
+
+    return sum(iouScores) / len(iouScores)
+
+
+def iouScoreImages(testMasks: list[np.ndarray], predictedMasks: list[np.ndarray], samples: list[ImageSample]) -> list[float]:
+    iouScores: list[float] = []
+    for testMask, predictedMask, sample in zip(testMasks, predictedMasks, samples):
+        iou = iouScoreImage(testMask, predictedMask)
+        iouScores.append(iou)
+        logging.info(f">> [Image Segmentation] IoU Score for sample {sample.name} (sample id: {sample.id}) is: {round(iou, 2)}")
+
+    return iouScores
+
+
+def addPadding(predictedMasks: list[np.ndarray], testMasksPadding: list[tuple[int, int]]) -> list[np.ndarray]:
```

Comment:
@VukManojlovic isn't this function already defined in coretexpylib?

Reply:
There is a resizeWithPadding function that accepts an image and the desired shape, and then pads the image to that shape. This function sets the padded area of the segmentation output to 0 in case the model's output overlaps that padded area.
I would suggest a function name change.